### PR TITLE
[fix] Include shared objects while filtering serilaizer fields #247

### LINKED ISSUE #247: [Bug] Include the objects with org=None, while filtering serilaizer fileds based on the organization of an user
https://github.com/openwisp/openwisp-users/blob/747ac34ac59c7fee843f8ad250cc5d12292c722b/openwisp_users/api/mixins.py#L137-L140

@nemesisdesign , @atb00ker with this line of code we are trying to filter out other `ForeignKey` related field other than `organization` field, based on the organization of a user, But mostly all of such fields has an option to create objects as shared org(organization=None). eg.:- templates, CA, etc.. But this LOC only returns the object belonging to an organization, excluding the objects whose organization is set to None.

Will it not be better, if we include the shared objects along with these.

CC: @Purhan

### PR DESCRIPTION
**Pull request by ManishShah120**
Closes #247

### CODE REVIEW

**Comment by ManishShah120** [comment]
@nemesisdesign , @atb00ker In the testapp no instance with relating was allowed to be created with `org=None`, so for writting tests for these changes I had to allow `Shelf` instances to be created with `org=None`, so therefore, I edited the existing migration file in the testapp.

File: openwisp_users/api/mixins.py [hunk 1]
```diff
@@ -117,6 +118,7 @@ class FilterSerializerByOrganization:
     """
 
     organization_lookup = 'organization__in'
+    filter_shared = True
```
Comment (nemesisdesign) [inline openwisp_users/api/mixins.py:121]: let's set it to `False` by default please, so we have to enable it explicitly on models which inherit `ShareableOrgMixin`

**Review by nemesisdesign — CHANGES_REQUESTED** [review]
Great progress Manish but seems to me you forgot to update the docs.

Please always make a habit to update documentation when already documented features are changed.

The docs for these serializers are here: https://github.com/openwisp/openwisp-users#multi-tenant-serializers-for-the-browsable-web-ui

Just mention this new attribute and provide an example.

See also my comment below.

File: openwisp_users/api/mixins.py [hunk 2]
```diff
@@ -135,8 +137,11 @@ def filter_fields(self):
                 )
                 continue
             try:
+                conditions = Q(**{self.organization_lookup: organization_filter})
```
Comment (atb00ker) [inline openwisp_users/api/mixins.py:140]: `conditions`, can we change the varible name to something like `org_filter` or `org_filter_condiitions` or something like that, `conditions` alone sounds out of place!

**Review by atb00ker — COMMENTED** [review]
Looks good!

File: openwisp_users/api/mixins.py (hunk 2 above)
Comment (nemesisdesign) [reply openwisp_users/api/mixins.py:140]: `conditions` is fine, those are the conditions of the filter so it's good enough for me

**Review by nemesisdesign — APPROVED** [review]

**Pull request merged**